interop-cats/jvm/src/main/scala/scalaz/zio/interop/catszmanagedjvm.scala

```diff
@@ -0,0 +1,109 @@
+package scalaz.zio.interop
+
+import cats.effect.Resource.{ Allocate, Bind, Suspend }
+import cats.effect.{ ExitCase, LiftIO, Resource, Sync, IO => CIO }
+import cats.{ effect, Bifunctor, Monad, MonadError, Monoid, Semigroup, SemigroupK }
+import scalaz.zio.{ Reservation, ZIO, ZManaged }
+import scalaz.zio.Runtime
+
+trait CatsZManagedSyntax {
+
+  implicit class ResourceSyntax[A](resource: Resource[CIO, A]) {
+    def toManaged[R, E](
+      implicit l: LiftIO[ZIO[R, Throwable, ?]]
+    ): ZManaged[R, Throwable, A] = {
+      def convert[A1](resource: Resource[CIO, A1]): ZManaged[R, Throwable, A1] =
+        resource match {
+          case Allocate(res) =>
+            ZManaged.unwrap(
+              l.liftIO(res)
+                .map {
+                  case (a, r) =>
+                    ZManaged.reserve(Reservation(ZIO.succeed(a), l.liftIO(r(ExitCase.Completed)).orDie.uninterruptible))
```

Comment:
I think `orDie` is correct for release, because in `ZIO`, failures in finalizers are treated as defects.

Reply:
Ok. Then I'll keep it as is. Might be a bit surprising depending on how common that is when using cats-effect. I'll add some documentation mentioning it